core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -1,13 +1,65 @@
 package com.mesosphere.usi.core
 
-import com.mesosphere.usi.core.models.PodSpec
+import akka.NotUsed
+import akka.stream.BidiShape
+import akka.stream.scaladsl.{BidiFlow, Broadcast, GraphDSL}
+import com.mesosphere.usi.models.{Mesos, SpecEvent, USIStateEvent}
 
-import scala.concurrent.Future
 
-class Scheduler {
+/*
+ * Provides the scheduler graph component. The component has two inputs, and two outputs:
+ *
+ * Input:
+ * 1) SpecEvents - Used to replicate the specification state from the framework to the USI scheduler
+ * 2) MesosEvents - Events from Mesos; offers, task status updates, etc.
+ *
+ * Output:
```

Comment:
If I'm not mistaken, `USIStateEvents` and `MesosCalls` collectively make up a `Frame` here. If so it would be nice to depict that in the diagram and in the writeup here.

Reply:
They make up a FrameResult, or, all of the effects since processing some incoming event (spec updated event or mesos event)